src/JZSolver.cpp

```diff
@@ -0,0 +1,106 @@
+#include "JZSolver.hpp"
+#include <numeric>
+#include <iostream>
+
+bool JZSolver::solve(std::shared_ptr<Board> b)
+{
+    verifyBoardClues(b);
+    findAndFillCompleteLines(b);
+
+    while(not b->isSolved())
+    {
+        stepCounter_++;
+        // methods to solve puzzle
+        // if method changes something at board then start loop from beginning
+        // if all methods haven't changed anything then break loop and admit failure
+        break;
+    }
+    return b->isSolved();
+}
+
+unsigned int JZSolver::getStepCounter()
+{
+    return stepCounter_;
+}
+
+void JZSolver::verifyBoardClues(std::shared_ptr<Board> b)
+{
+    if(b->getCluesCols().size() != b->getSizeCols())
+        throw std::length_error("Invalid clues: some columns are missing clues.");
+
+    if(b->getCluesRows().size() != b->getSizeRows())
+        throw std::length_error("Invalid clues: some rows are missing clues.");
+
+    unsigned int sumRows = 0;
+    unsigned int sumCols = 0;
+
+    for(auto row : b->getCluesRows())
+    {
+        sumRows += std::accumulate(row.begin(), row.end(), 0);
+    }
+
+    for(auto col : b->getCluesCols())
+    {
+        sumCols += std::accumulate(col.begin(), col.end(), 0);
+    }
+
+    if(sumRows != sumCols)
+    {
+        std::string msg = "Invalid clues: different object count in rows (which is " +
+                          std::to_string(sumRows) + ") and columns (which is " +
+                          std::to_string(sumCols) + ").";
+        throw std::logic_error(msg);
+    }
+}
+
+void JZSolver::findAndFillCompleteLines(std::shared_ptr<Board> b)
+{
+#if DEBUG
+    unsigned int lineCounter = 0;
+#endif
+    for(int rowCnt = 0; rowCnt < b->getSizeRows(); rowCnt++)
+    {
+        auto row =  b->getCluesRows()[rowCnt];
+        unsigned int cluesResult = std::accumulate(row.begin(), row.end(), 0) + row.size() - 1;
```

Comment:
you can extract a new variable for better clarity:
`auto spaces = row.size() - 1;`

Reply:
done